packages/devextreme/js/__internal/ui/splitter/resize_handle.ts

```diff
@@ -45,12 +55,42 @@ class ResizeHandle extends (Widget as any) {
   _initMarkup(): void {
     super._initMarkup();
 
-    this._toggleDirectionClass();
-    this.$element().addClass(RESIZE_HANDLE_CLASS);
+    this._renderResizeHandleContent();
 
     this._setAriaAttributes();
   }
 
+  _renderResizeHandleContent(): void {
+    this.$element().addClass(RESIZE_HANDLE_CLASS);
+    this._toggleDirectionClass();
+
+    this._$collapsePrevButton = $('<div>').addClass(this._getCollapseIconClass(false)).appendTo(this.$element());
+    this._$resizeHandle = $('<div>').addClass(RESIZE_HANDLE_ICON_CLASS).appendTo(this.$element());
+    this._$collapseNextButton = $('<div>').addClass(this._getCollapseIconClass(true)).appendTo(this.$element());
+
+    this._setResizeHandleContentVisibility();
+  }
+
+  _getCollapseIconClass(isNextButton: boolean): string {
+    const isHorizontal = this._isHorizontalDirection();
+    let classList = `${isNextButton ? RESIZE_HANDLE_COLLAPSE_NEXT_BUTTON_CLASS : RESIZE_HANDLE_COLLAPSE_PREV_BUTTON_CLASS} ${ICON_CLASS}`;
+    if (isNextButton) {
```

Comment:
I think we can extract logic that will return right direction 'right', 'left', 'down', 'up' according to orientation in separate method